wcfsetup/install/files/lib/system/trophy/condition/TrophyConditionHandler.class.php

```diff
@@ -85,6 +85,30 @@ public function assignTrophies($maxAssigns = 500) {
 		}
 	}
 	
+	/**
+	 * Revoke user trophies which are not longer fulfills the conditions. 
+	 * 
+	 * @param 	integer		$maxRevokes
+	 */
+	public function revokeTrophies($maxRevokes = 500) {
+		$trophyList = new TrophyList();
+		$trophyList->getConditionBuilder()->add('awardAutomatically = ?', [1]);
+		$trophyList->getConditionBuilder()->add('revokeAutomatically = ?', [1]);
+		$trophyList->getConditionBuilder()->add('isDisabled = ?', [0]);
+		$trophyList->readObjects();
+		
+		$i = 0;
+		foreach ($trophyList as $trophy) {
+			$userTrophyIDs = $this->getRevocableUserTrophyIDs($trophy);
```

Comment:
You should pass `$maxRevokes - $i` into this function and add it as a limit to the SELECT in there.